Lib/sqlite3/test/hooks.py

```diff
@@ -248,6 +249,27 @@ def trace(statement):
                         "Unicode data %s garbled in trace callback: %s"
                         % (ascii(unicode_value), ', '.join(map(ascii, traced_statements))))
 
+    @unittest.skipIf(sqlite.sqlite_version_info < (3, 3, 9), "sqlite3_prepare_v2 is not available")
+    def CheckTraceCallbackContent(self):
+        """
+        Test that the statement are correct. Fix for bpo-26187
+        """
+        with NamedTemporaryFile(suffix='.sqlite') as db_path:
+            traced_statements = []
+            def trace(statement):
+                traced_statements.append(statement)
+
+            queries = ["create table foo(x)",
```

Comment:
`traced_statements`, `trace` and `queries` can go outside of the with statements.